src/agents/gc/OSConfig_PrototypeResource.c

```diff
@@ -0,0 +1,659 @@
+        }
+
+        memset(&miValue, 0, sizeof(miValue));
+    }
+    
+    // Set the created output resource instance as the output resource in the GetTargetResource instance
+    if (MI_RESULT_OK != (miResult = MI_Instance_SetElement(&get_result_object.__instance, MI_T("OutputResource"), &miValueResource, MI_INSTANCE, 0)))
+    {
+        OsConfigLogError(GetLog(), "GetTargetResource: MI_Instance_SetElement(OutputResource) failed with %d", miResult);
+        goto Exit;
+    }
+    else
+    {
+        OsConfigLogError(GetLog(), "GetTargetResource: OSConfig_PrototypeResource_GetTargetResource_Post failed with %d", miResult);
+    }
+        
+Exit:
+    if (MI_RESULT_OK != miResult)
+    {
+        MI_Context_PostError(context, miResult, MI_RESULT_TYPE_MI, 0);
+    }
+
+    // Clean up the Result MI value instance if needed
+    if ((NULL != miValueResult.instance) && (MI_RESULT_OK != MI_Instance_Delete(miValueResult.instance)))
+    {
+        OsConfigLogInfo(GetLog(), "GetTargetResource: MI_Instance_Delete(miValueResult) failed");
+    }
+
+    // Clean up the output resource instance
+    if ((NULL != resultResourceObject) && (MI_RESULT_OK != MI_Instance_Delete(resultResourceObject)))
+    {
+        OsConfigLogInfo(GetLog(), "GetTargetResource: MI_Instance_Delete(resultResourceObject) failed");
+    }
+
+    // Clean up the GetTargetResource instance
+    if (MI_RESULT_OK != (miResult == OSConfig_PrototypeResource_GetTargetResource_Destruct(&get_result_object)))
+    {
+        OsConfigLogError(GetLog(), "GetTargetResource: OSConfig_PrototypeResource_GetTargetResource_Destruct failed with %d", miResult);
+    }
+
+    // Post MI result back to MI to finish
+    OsConfigLogInfo(GetLog(), "GetTargetResource: complete");
+    MI_Context_PostResult(context, miResult);
+}
+
+void MI_CALL OSConfig_PrototypeResource_Invoke_TestTargetResource(
+    _In_opt_ OSConfig_PrototypeResource_Self* self,
+    _In_ MI_Context* context,
+    _In_opt_z_ const MI_Char* nameSpace,
+    _In_opt_z_ const MI_Char* className,
+    _In_opt_z_ const MI_Char* methodName,
+    _In_ const OSConfig_PrototypeResource* resourceClass,
+    _In_opt_ const OSConfig_PrototypeResource_TestTargetResource* in)
+{
+    MI_UNREFERENCED_PARAMETER(self);
+    MI_UNREFERENCED_PARAMETER(resourceClass);
+
+    OSConfig_PrototypeResource_TestTargetResource test_result_object;
+
+    MI_Result miResult = MI_RESULT_OK;
+    MI_Boolean is_compliant = MI_FALSE;
+
+    MI_Value miValueResult;
+    memset(&miValueResult, 0, sizeof(MI_Value));
+
+    OsConfigLogInfo(GetLog(), "TestTargetResource: start (namespace %s, class name %s, method name %s)", nameSpace, className, methodName);
+
+    if ((in == NULL) || (in->InputResource.exists == MI_FALSE) || (in->InputResource.value == NULL))
+    {
+        OsConfigLogError(GetLog(), "TestTargetResource: invalid OSConfig_PrototypeResource_TestTargetResource argument");

```

Comment:
>OsConfigLogError(GetLog(), "TestTargetResource: invalid OSConfig_PrototypeResource_TestTargetResource argument"); [](http://example.com/codeflow?start=8&length=113)

it will be nice to report these errors to gcagent too so that gcagent will collect it and report it to cloud service, currently you are posting error at line 520 but not specifying the error message (4th param is 0)

Reply:
Q: how to report (or log) these for the GC agent? Are there any other ways except mentioned error message (that is only one, and these are traces meant to be used as a sequence). Is there a GC log we can write too, sequentially?

Reply:
Great, thanks!